velox/functions/sparksql/RegexFunctions.cpp

```diff
@@ -57,6 +82,310 @@ void ensureRegexIsCompatible(
+  /// written to the resultRef vector.
+  ///
+  /// @param rows The rows of data to be processed.
+  /// @param args The arguments passed to the function.
+  ///             args[0]: The input strings.
+  ///             args[1]: The pattern to search for.
+  ///             args[2]: The replacement strings.
+  ///             args[3]: (Optional) The starting position.
+  /// @param context The evaluation context.
+  /// @param resultRef The vector to write the result to.
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& /*outputType*/,
+      exec::EvalCtx& context,
+      VectorPtr& resultRef) const final {
+    VELOX_CHECK(args.size() == 3 || args.size() == 4);
+
+    exec::LocalDecodedVector input(context, *args[0], rows);
+
+    BaseVector::ensureWritable(
+        rows, ARRAY(VARCHAR()), context.pool(), resultRef);
+    exec::VectorWriter<Array<Varchar>> resultWriter;
+    resultWriter.init(*resultRef->as<ArrayVector>());
+
+    exec::LocalDecodedVector overwriteVec(context, *args[2], rows);
+
+    int64_t position;
+    exec::LocalDecodedVector positionVec(context);
+    if (args.size() == 4) {
+      positionVec = exec::LocalDecodedVector(context, *args[3], rows);
+    }
+
+    const re2::RE2 re_(basePattern_, RE2::Quiet);
+
+    try {
+      checkForBadPattern(re_);
+      checkForCompatiblePattern(basePattern_);
+    } catch (const std::exception& e) {
+      context.setErrors(rows, std::current_exception());
+      return;
+    }
+
+    bool mustRefSourceStrings = false;
+
+    rows.applyToSelected([&](vector_size_t row) {
+      resultWriter.setOffset(row);
+      auto& arrayWriter = resultWriter.current();
+
+      position = 1;
+      if (args.size() == 4) {
+        position = positionVec->valueAt<int64_t>(row);
+      }
+      VELOX_CHECK(!(position - 1 < 0));
+
+      const StringView& currentOverwrite =
+          overwriteVec->valueAt<StringView>(row);
+
+      // For an unknown reason, when we do not create the StringPiece with
+      // .getString() and instead reference the StringView, the line below
+      // "strValue.substr" can overwrite the values found in overwrite. Cannot
+      // explain this behavior and must default to getString
+      std::vector<re2::StringPiece> overwrite = {
+          re2::StringPiece(currentOverwrite.getString())};
+
+      const StringView& current = input->valueAt<StringView>(row);
+      std::string strValue = current.getString();
+
+      // If position is not 1, adjust the string for regex operation.
+      // Spark implementation default position is 1. It is 1-indexed instead of
+      // 0-indexed.
+      std::string prefix;
+      if (position - 1 > 0) {
+        prefix = strValue.substr(0, position - 1);
+        if (position - 1 > strValue.length()) {
+          strValue = "";
+        } else {
+          strValue = strValue.substr(position - 1);
+        }
```

Comment:
@mbasmanova I implemented the changes that you have requested. All tests passed locally, but I did include changes that allow for the constant pattern use case to be called on multiple rows at once which was left out as an oversight. 

In doing so I ran into an issue that I previously ran into with the non-constant pattern use case. The issue and relevant code are highlighted in the code referenced. Previously, overwrite was declared using just StringView, but debugging revealed that after the .substr line is run, overwrites buffer gets overwritten. This behavior does not occur when .getString() is used to create the StringPiece.

Do you have an idea as to why this is happening?